src/graphql/mutation/braintreeCreateMonthlyGoodSubscription.graphql

```diff
@@ -1,4 +1,4 @@
-mutation braintreeCreateMonthlyGoodSubscription($paymentMethodNonce: String!, $amount: Money!, $donateAmount: Money!, $dayOfMonth: Int!, $category: MonthlyGoodCategoryEnum, $isOnetime: Boolean) {
+mutation braintreeCreateMonthlyGoodSubscription($paymentMethodNonce: String!, $deviceData: String!, $amount: Money!, $donateAmount: Money!, $dayOfMonth: Int!, $category: MonthlyGoodCategoryEnum, $isOnetime: Boolean) {
```

Comment:
Small item, deviceData is not a required field in the graphql mutation so you can drop the `!`.

If you want to keep it required here you should check for it alongside the nonce before calling the mutation.